source/dotnet/Test/AdaptiveCards.Test/AdaptiveActionTests.cs

```diff
@@ -29,6 +29,39 @@ public void ParseIconUrl()
             Assert.AreEqual(expectedIconUrl, actualIconUrl);
         }
 
+        [TestMethod]
+        public void ParseInvalidAction()
+        {
+            string url = "http://adaptivecards.io/content/cats/1.png";
+            var json = @"{
+	""$schema"": ""http://adaptivecards.io/schemas/adaptive-card.json"",
+	""type"": ""AdaptiveCard"",
+	""version"": ""1.0"",
+	""body"": [
+		{
+			""type"": ""Image"",
+			""url"": """+ url + @""",
+			""selectAction"": {
```

Comment:
What happens if there's a property placed AFTER selectAction? Please try the following

```json
{
  "type": "Image",
  "selectAction": {
    "type": "InvalidAction"
  },
  "url": "https://msn.com"
}
```

And validate that the Url property gets populated correctly.

Reply:
Yeah. Mentioned in original comment. It drops the url. The problem is that the bug is in JSON.NET.

Reply:
Ah I see, so it's intended to only be a partial fix. Makes sense.

Reply:
Does changing the version of JSON.NET address the issue? That might be another way to handle this.

---
In reply to: [232431897](https://github.com/Microsoft/AdaptiveCards/pull/2163#discussion_r232431897) [](ancestors = 232431897)